services/blockchain-indexer/shared/utils/eventsUtils.js

```diff
@@ -76,21 +74,29 @@ const getEventsInfoToIndex = async (block, events) => {
 	return eventsInfoToIndex;
 };
 
-const deleteEventsTillHeight = async (blockHeight, dbTrx) => {
+const deleteEventsTillFinalizedHeight = async () => {
```

Comment:
```suggestion
const deleteEventStrTillFinalizedHeight = async () => {
```

Also, this entire logic should be part of an explicit MySQL transaction, including the KV store updates.